SQRLDotNetClientUI/ViewModels/ReKeyViewModel.cs

```diff
@@ -0,0 +1,297 @@
+            });
+
+
+        }
+
+        /// <summary>
+        /// Checks if the password verification matches the new password
+        /// and enables/disables UI controls accordingly.
+        /// </summary>
+        private void CheckPasswordVerification()
+        {
+            if (!string.IsNullOrEmpty(this.NewPassword) && this.NewPassword == this.NewPasswordVerify)
+            {
+                this.CanSave = true;
+            }
+            else this.CanSave = false;
+        }
+
+        public void Cancel()
+        {
+            ((MainWindowViewModel)_mainWindow.DataContext).Content =
+                ((MainWindowViewModel)_mainWindow.DataContext).MainMenu;
+        }
+
+        /// <summary>
+        /// Runs the Re-Key Logic
+        /// </summary>
+        public async void Next()
+        {
+            //Label used for retrying the rescue code if you type it wrong the first time
+            RetryRescueCode:
+
+            //Dialog Box used to capture the user's current Rescue Code
+            InputSecretDialogView rescueCodeDlg = new InputSecretDialogView(SecretType.RescueCode);
+            rescueCodeDlg.WindowStartupLocation = WindowStartupLocation.CenterOwner;
+            string rescueCode = await rescueCodeDlg.ShowDialog<string>(
+                _mainWindow);
+
+            //List of Progress Reporters to be used during the re-key process (progress bar magic)
+            var  progressList = new List<Progress<KeyValuePair<int, string>>>(){ new Progress<KeyValuePair<int, string>>(),new Progress<KeyValuePair<int, string>>() };
+
+            //Progress Dialog will show our "Progress" as the Identity is Decrypted, and Re-Encrypted for Rekey
+            var progressDialog = new ProgressDialog(progressList);
+            progressDialog.HideFinishedItems = true;
+            progressDialog.HideEnqueuedItems = true;
+            _ = progressDialog.ShowDialog(_mainWindow);
+
+            //Actually do the Re-Key Work
+            var result = await SQRL.RekeyIdentity(_identityManager.CurrentIdentity, SQRL.CleanUpRescueCode(rescueCode), NewPassword, progressList[0],progressList[1]);
+            if(!result.Success)
+            {
+                progressDialog.Close();
+                //Fail bad rescue code (something went wrong...) try again?
+                var answer = await new Views.MessageBox(_loc.GetLocalizationValue("ErrorTitleGeneric"),
+                                                                   _loc.GetLocalizationValue("InvalidRescueCodeMessage"),
+                                                                   MessageBoxSize.Small, MessageBoxButtons.YesNo, MessageBoxIcons.ERROR)
+                                                                   .ShowDialog<MessagBoxDialogResult>(_mainWindow);
+                if (answer == MessagBoxDialogResult.YES)
+                {
+                    goto RetryRescueCode; //Go back up and re-do it all, this time with passion!
```

Comment:
LOL @ "this time with passion!", what a great comment! 🤣
Makes reviewing the code a lot more pleasant. 😀